src/Schema/Blueprint.php

```diff
@@ -16,6 +16,91 @@
 use function is_string;
 use function key;
 
+/**
+ * @link https://www.mongodb.com/docs/atlas/atlas-search/define-field-mappings/#std-label-fts-field-mappings
+ * @phpstan-type TypeSearchIndexField array{
+ *     type: 'boolean'|'date'|'dateFacet'|'objectId'|'stringFacet'|'uuid',
+ * } | array{
+ *     type: 'autocomplete',
+ *     analyzer?: string,
+ *     maxGrams?: int,
+ *     minGrams?: int,
+ *     tokenization?: 'edgeGram'|'rightEdgeGram'|'nGram',
+ *     foldDiacritics?: bool,
+ * } | array{
+ *     type: 'document'|'embeddedDocuments',
+ *     dynamic?:bool,
+ *     fields: array<string, array<mixed>>,
+ * } | array{
+ *     type: 'geo',
+ *     indexShapes?: bool,
+ * } | array{
+ *     type: 'number'|'numberFacet',
+ *     representation?: 'int64'|'double',
+ *     indexIntegers?: bool,
+ *     indexDoubles?: bool,
+ * } | array{
+ *     type: 'token',
+ *     normalizer?: 'lowercase'|'none',
+ * } | array{
+ *     type: 'string',
+ *     analyzer?: string,
+ *     searchAnalyzer?: string,
+ *     indexOptions?: 'docs'|'freqs'|'positions'|'offsets',
+ *     store?: bool,
+ *     ignoreAbove?: int,
+ *     multi?: array<string, array<mixed>>,
```

Comment:
The [multi example](https://www.mongodb.com/docs/atlas/atlas-search/field-types/string-type/#try-an-example-for-the-fts-field-type-type) suggests this would be a map of strings to string field definitions. Since those don't exist, I suppose you could use `TypeSearchIndexField`, but maybe not possible since it's recursive or undesirable as it would suggest that users could use _other_ field definition types.

Reply:
The example looks like a hashmap of hashmap. I can't use the type `TypeSearchIndexField` recursively.
```json
        "multi": {
          "english": {
            "type": "string",
            "analyzer": "lucene.english"
          },
```